### DISCUSSION TITLE
SOQL does not allow complex queries across related objects.

### DISCUSSION THREAD

**Discussion by JohnRCoder** [forcedotcom/salesforcedx-vscode#5479] [Ideas]
**In Workbench, I can query across related objects**
Example. Select Id, Account.Id, Contact.Id From Case where Contact.Email Like '%@specialdomain.com'

**We should be able to do the same query in VS Code SOQL**
The second you try to type a period to select sub-object.field, all the drop downs stop working.

**Thank you**
Much appreciated if you can look into and implement.

**Comment by JohnRCoder** [comment]
Any idea when this is going to be worked on?

**Comment by mshanemc** [comment]
try it now!

**Comment by daphne-sfdc** [comment]
We are going to add support for relationship queries and subqueries soon 🙂 

GUS WI here: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002Yizn9YAB/view
